openvdb/Grid.h

```diff
@@ -1516,9 +1516,7 @@ template<typename TreeT>
 inline void
 Grid<TreeT>::pruneGrid(float tolerance)
 {
-    OPENVDB_NO_TYPE_CONVERSION_WARNING_BEGIN
-    const auto value = zeroVal<ValueType>() + tolerance;
-    OPENVDB_NO_TYPE_CONVERSION_WARNING_END
+    const auto value = math::cwiseAdd(zeroVal<ValueType>(), tolerance);
     this->tree().prune(static_cast<ValueType>(value));
 }
```

Comment:
One other option I've been considering is to simply add explicit functionality to Grid::prune which branches on matrix types - something like:
```
auto addTolerance(auto& a, float tol) { return a + tol; }

template<size_t SIZE, typename T>
auto addTolerance(Mat<SIZE, T>& a, float tol) { // do cwiseadd }

prune(float a)
{
    auto value = addTolerance(zeroVal<ValueType>(), a);
    ....
}
```
I think I'd actually prefer this, if we assume the following:

 - `cwiseAdd` is only necessary as we are still debating if/what `operator+` should perform on matrix types with scalars
 - `Grid::prune` itself is to be deprecated.